src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -683,6 +684,7 @@ Shard::initSQLite(std::lock_guard<std::recursive_mutex> const&)
 
         if (backendComplete_)
         {
+            setup.noPragma();
```

Comment:
Maybe initialize `setup` so it's `const` and not change the `setup` object in the arms of the if statement? Something like:
```
    DatabaseCon::Setup const setup = [&] {
        DatabaseCon::Setup result;
        result.startUp = config.START_UP;
        result.standAlone = config.standalone();
        result.dataDir = dir_;
        if (backendComplete_)
            result.noPragma();
        else
            result.usePragma();
        return result;
    }();
```

Reply:
Done